---
title: "Day 1 - Learning objectives"
teaching: 5
exercises: 0
questions:
- "What is RNA-Seq?"
- "What is the case-study for this workshop?"
- "How to use bash terminal?"
- "What are the steps in RNA-seq differential expression analysis?"
- "Nextfow workflows for Bioinformatics: Why bother?"
- "How to run 'nfcore-rnaseq' on the Nimbus instance?"
- "What are the imporatant nfcore-rnaseq inputs and outputs?"

objectives:
- "Quick overview of the primary learning objectives for DAY1"

keypoints:
- XXX
---

#### RNA-Seq
```
- What is RNA-Seq?
- What are the main applications of RNA-Seq
- How does a basic RNA-seq workflow look like?
- What is the experimental case-study for this workshop?
```

#### A specific case-study using RNA-Seq
```
. Knockout mouse model to study Williams-Beuren Syndrome (WBS)
. Experimental design
. Experimental question
```

#### List the steps involved in analysis of RNA-seq data
```
- Check sequencing read quality
- Trimming seqeuncing reads 
- Alignment against reference and visualisation
- Differential Expression Analysis
```

#### Describe the benefits of using nf-core workflows
```
- A typical bioinformatics workflow - too complicated?
- Nextflow as a workflow management systems, how does it help?
- Use nfcore’s rnaseq pipeline to transform raw sequencing data into analysis ready count data.
```

#### Deploy an RNA-seq nf-core workflow on Pawsey’s Nimbus VM
```
- What is the core nfcore-rnaseq command?
- What are the main inputs?
- What are the main outputs? 
- Can we add/remove/change the tools/parameters used for the various processes?
- What do we carry forward to DAY 2: The gene-count matrix
```

___
**Notes**   
<sup id="f1">1[↩](#a1)</sup> Here is a list of ['nfcore pipelines'](https://nf-co.re/pipelines/).

___
<br>



{% include links.md %}
