---
title: "Raw data and quality checking"
teaching: 5
exercises: 10
questions:
- "Why trim RNA seqeuncig reads?"

objectives:
- "Understand the requirement for trimming RNA sequencing reads"
- "Understand the pros and cons of trimming"

keypoints:
- 
- 
---

- Trimming is sometimes performed to improve the quality of the raw data and potentially improve its mappability. There are several ways to perform trimming:
  - Removal of poor quality reads or bases (e.g. ends of reads)
  - Removal of adapter sequences
  - Removal of polyA tails

__NOTE:__
- Be very wary about trimming RNA sequencing data
  - Trimming of poor quality reads can affect gene expression estimates (Williams at el., 2016)

- Trimming of high quality adapter sequences was shown to increase quality and reliability of biological signals in RNA-seq data (Dozmorov et al., 2015)


__CHALLENGE__
- From looking into the results folder, can you tell :
  - Which tool does nfcore-rnaseq use for read-trimming?
  - Was there an effect of the trimming step on the sample? Choose one sample e.g. SRR3473989
    - Which tool did you use for visualisation of the quality? Before and after trimming?
    - Which folders in the 'results' did you need to access? 
  

